src/Actions/Commission/CreateCommissionCreditNotes.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace FluxErp\Actions\Commission;
+
+use FluxErp\Actions\FluxAction;
+use FluxErp\Actions\Order\CreateOrder;
+use FluxErp\Actions\OrderPosition\CreateOrderPosition;
+use FluxErp\Enums\OrderTypeEnum;
+use FluxErp\Models\Commission;
+use FluxErp\Models\Order;
+use FluxErp\Models\OrderPosition;
+use FluxErp\Models\OrderType;
+use FluxErp\Models\User;
+use FluxErp\Models\VatRate;
+use FluxErp\Rulesets\Commission\CreateCommissionCreditNotesRuleset;
+use FluxErp\Support\Collection\OrderCollection;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+class CreateCommissionCreditNotes extends FluxAction
+{
+    protected function boot(array $data): void
+    {
+        parent::boot($data);
+
+        $this->rules = resolve_static(CreateCommissionCreditNotesRuleset::class, 'getRules');
+    }
+
+    public static function models(): array
+    {
+        return [Commission::class];
+    }
+
+    public function performAction(): OrderCollection
+    {
+        $agents = User::query()
+            ->withWhereHas('contact', fn (Builder|BelongsTo $query) => $query->with('invoiceAddress.country'))
```

Comment:
validate if the user even has a contact? Because it is not guaranteed.

Reply:
thats the point of withWhereHas, only users who have a linked contact are returned and the contact is eager loaded too